### PR TITLE
chore(anomaly detection): feedback banner

### PR DESCRIPTION
**Pull request by mifu67**
https://github.com/user-attachments/assets/d2d96f24-9771-48a4-b587-3f25372c6bef

### CODE REVIEW

File: static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  if (isSubmitted) {
+    return null;
+  }
+  return (
+    <StyledAlert
+      type="info"
+      trailingItems={
+        <Fragment>
+          <StyledButton onClick={() => handleClick(true)}>Yes</StyledButton>
+          <StyledButton onClick={() => handleClick(false)}>No</StyledButton>
```
Comment (ameliahsu) [inline static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.tsx:81]: Yes and No should be translated (`t('Yes')` and `t('No')`)

File: static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  z-index: 1;
+`;
+
+CloseButton.defaultProps = {
+  icon: <IconClose />,
+  ['aria-label']: t('Close'),
+  priority: 'link',
+  borderless: true,
+  size: 'xs',
+};
```
Comment (natemoo-re) [inline static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.tsx:106]: No longer used?

File: static/app/views/alerts/rules/metric/details/body.tsx [hunk 3]
```diff
@@ -236,13 +237,22 @@ export default function MetricDetailsBody({
...
+            />
+          )}
+
           <ErrorMigrationWarning project={project} rule={rule} />
 
           {/* TODO: add activation start/stop into chart */}
           <MetricChart
             api={api}
             rule={rule}
-            incidents={incidents}
```
Comment (natemoo-re) [inline static/app/views/alerts/rules/metric/details/body.tsx:245]: Intentional deletion?

File: static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        incident_id: mockIncident.id,
+      })
+    );
+
+    expect(feedbackClient.captureEvent).toHaveBeenCalledWith({
+      message: 'Anomaly Detection Alerts Banner Feedback',
+      level: 'info',
+      tags: {
+        featureName: 'anomaly-detection-alerts-feedback',
+        choice_selected: true,
```
Comment (natemoo-re) [inline static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.spec.tsx:53]: Nice! Want to add a test for the `No` behavior, too?

**Review by natemoo-re — COMMENTED** [review]
Looking great!

File: static/app/views/alerts/rules/metric/details/body.tsx (hunk 3 above)
Comment (mifu67) [reply static/app/views/alerts/rules/metric/details/body.tsx:245]: Definitely not and I was confused about why the incidents weren't rendering! Thanks for catching this 😅

File: static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.tsx (hunk 2 above)
Comment (mifu67) [reply static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.tsx:106]: Yes, I checked the designs and there's no close button so I removed it (easier to style without the close button). I'll get rid of this code.

File: static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+export default function AnomalyDetectionFeedbackBanner({
+  id,
+  organization,
+  selectedIncident,
+}: AnomalyDetectionFeedbackProps) {
+  const [isSubmitted, submit] = useDismissable(id);
+
+  const handleClick = useCallback(
+    (isAnomaly: boolean) => {
```
Comment (natemoo-re) [inline static/app/views/alerts/rules/metric/details/anomalyDetectionFeedbackBanner.tsx:28]: Not blocking, but this variable name is unclear! Maybe `anomalyCorrectlyIdentified`?

File: static/app/views/alerts/rules/metric/details/body.tsx [hunk 6]
```diff
@@ -236,6 +237,16 @@ export default function MetricDetailsBody({
             )}
           </StyledSubHeader>
 
+          {selectedIncident?.alertRule.detectionType ===
+            AlertRuleComparisonType.DYNAMIC && (
+            <AnomalyDetectionFeedbackBanner
+              key={selectedIncident.id}
```
Comment (natemoo-re) [inline static/app/views/alerts/rules/metric/details/body.tsx:243]: Comment to note why `key` is important here would be helpful for future visitors.

```suggestion
              // unique key to force re-render when incident changes
              key={selectedIncident.id}
```

**Review by natemoo-re — APPROVED** [review]
Nice work! Consider both comments here as non-blocking suggestions. I would be comfortable with this getting merged as-is!

**Pull request merged**